tracer/src/Datadog.Trace/Iast/Aspects/System/StringAspects.cs

```diff
@@ -512,6 +512,104 @@ public static string Concat<T>(IEnumerable<T> values)
+        }
+        catch (Exception ex)
+        {
+            IastModule.LogAspectException(ex, $"{nameof(StringAspects)}.{nameof(Join)}");
+        }
+
+        return result;
+    }
+
+    /// <summary>
+    /// String.Join aspect
+    /// </summary>
+    /// <param name="param1"> First param </param>
+    /// <param name="param2"> Second param </param>
+    /// <param name="param3"> Third param </param>
+    /// <returns> Join result </returns>
+    [AspectMethodReplace("System.String::Concat(System.ReadOnlySpan`1<System.Char>,System.ReadOnlySpan`1<System.Char>,System.ReadOnlySpan`1<System.Char>)")]
+    public static string Concat(ReadOnlySpan<char> param1, ReadOnlySpan<char> param2, ReadOnlySpan<char> param3)
+    {
+        var result = string.Concat(param1, param2, param3);
+        try
+        {
+            return StringModuleImpl.OnStringConcat(new StringConcatParams(TaintedObjects.GetString(ref param1), TaintedObjects.GetString(ref param2), TaintedObjects.GetString(ref param3)), result);
+        }
+        catch (Exception ex)
+        {
+            IastModule.LogAspectException(ex, $"{nameof(StringAspects)}.{nameof(Join)}");
+        }
+
+        return result;
+    }
+
+    /// <summary>
+    /// String.Join aspect
+    /// </summary>
+    /// <param name="param1"> First param </param>
+    /// <param name="param2"> Second param </param>
+    /// <param name="param3"> Third param </param>
+    /// <param name="param4"> Fourth param </param>
+    /// <returns> Join result </returns>
+    [AspectMethodReplace("System.String::Concat(System.ReadOnlySpan`1<System.Char>,System.ReadOnlySpan`1<System.Char>,System.ReadOnlySpan`1<System.Char>,System.ReadOnlySpan`1<System.Char>)")]
+    public static string Concat(ReadOnlySpan<char> param1, ReadOnlySpan<char> param2, ReadOnlySpan<char> param3, ReadOnlySpan<char> param4)
+    {
+        var result = string.Concat(param1, param2, param3, param4);
+        try
+        {
+            return StringModuleImpl.OnStringConcat(new StringConcatParams(TaintedObjects.GetString(ref param1), TaintedObjects.GetString(ref param2), TaintedObjects.GetString(ref param3), TaintedObjects.GetString(ref param4)), result);
+        }
+        catch (Exception ex)
+        {
+            IastModule.LogAspectException(ex, $"{nameof(StringAspects)}.{nameof(Join)}");
+        }
+
+        return result;
+    }
+
+    /// <summary>
+    /// String.Join aspect
+    /// </summary>
+    /// <param name="param"> Strings to concat </param>
+    /// <returns> Join result </returns>
+    [AspectMethodReplace("System.String::Concat(System.ReadOnlySpan`1<System.String>)")]
+    public static string Concat(ReadOnlySpan<string> param)
+#pragma warning disable DD0005 // Aspect is in incorrect format
+    {
+        // TODO: use propper overload when dotnet 9 or later target is available
+        var values = param.ToArray();
```

Comment:
How concerned should we be about the fact that this allocates an array, when the whole purpose of the `Concat(System.ReadOnlySpan`1<System.String>)` call is to _not_ allocate 🤔 

Somewhat unrelated, but I was trying to find the source code for the `Concat(System.ReadOnlySpan`1<System.String>)` call (to see if we could trivially implement it here instead) and couldn't find it anywhere 🤔